src/Ocelot/Logging/AspDotNetLogger.cs

```diff
@@ -11,69 +11,39 @@ public class AspDotNetLogger : IOcelotLogger
 
         public AspDotNetLogger(ILogger logger, IRequestScopedDataRepository scopedDataRepository)
         {
-            _logger = logger;
-            _scopedDataRepository = scopedDataRepository;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _scopedDataRepository = scopedDataRepository ?? throw new ArgumentNullException(nameof(scopedDataRepository));
```

Comment:
I don't know, but imho, we shouldn't have cases like these, expecting something from DI without checking.

Reply:
I'm OK too throw exceptions, but... It is strange because all injected services comes from DI-container, but who knows may be it is injected somehow differently.
So, if `AspDotNetLogger` is in DI already, it seems we don't need to generate exceptions, because DI container must throw exceptions.
Ah, I don't consider it as an issue if we throw exceptions, but could you check the case, please?

Reply:
No, not extra libraries, ```IRequestScopedDataRepository```  is a singleton in Ocelot Builder, and ```ILogger``` is instantiated in ```AspDotNetLoggerFactory```